packages/rlc-common/src/transformSampleGroups.ts

```diff
@@ -0,0 +1,420 @@
+    requestParameter.body.body?.length > 0
+  ) {
+    const body = requestParameter.body.body[0];
+    const bodyTypeName = body.typeName ?? body.type;
+    if (bodyTypeName !== "string" && body.oriSchema) {
+      schemaMap.set(bodyTypeName, body.oriSchema);
+    }
+    allSideAssignments.push(
+      ` body: ` + generateParameterTypeValue(bodyTypeName, "body", schemaMap)
+    );
+  }
+
+  requestParameter.parameters
+    ?.filter((p) => p.type === "query")
+    .forEach((p) => {
+      const name = `${p.name}`;
+      querySideAssignments.push(
+        `${name}: ` + generateParameterTypeValue(p.param.type, name, schemaMap)
+      );
+    });
+
+  if (querySideAssignments.length > 0) {
+    allSideAssignments.push(
+      ` queryParameters: { ` + querySideAssignments.join(", ") + `}`
+    );
+  }
+  requestParameter.parameters
+    ?.filter((p) => p.type === "header")
+    .filter((p) => p.name.toLowerCase() !== "contenttype")
+    .forEach((p) => {
+      const name = `${p.name}`;
+      headerSideAssignments.push(
+        `${name}: ` + generateParameterTypeValue(p.param.type, name, schemaMap)
+      );
+    });
+  if (headerSideAssignments.length > 0) {
+    allSideAssignments.push(
+      ` headers: { ` + headerSideAssignments.join(", ") + `}`
+    );
+  }
+  const contentType = requestParameter.parameters
+    ?.filter((p) => p.type === "header")
+    .filter((p) => p.name.toLowerCase() === "contenttype");
+  if (contentType && contentType.length > 0) {
+    allSideAssignments.push(
+      ` ${contentType[0].name}: ` +
+        generateParameterTypeValue(
+          contentType[0].param.type,
+          contentType[0].name,
+          schemaMap
+        )
+    );
+  }
+  let value: string = `{}`;
+  if (allSideAssignments.length > 0) {
+    value = `{ ` + allSideAssignments.join(", ") + `}`;
+  } else {
+    return [];
+  }
+
+  const optionParam: SampleParameter = {
+    name: "options",
+    assignment: `const options: ${method.optionsName} =` + value + `;`
+  };
+  addValueInImportedDict(packageName, method.optionsName, importedDict);
+  return [optionParam];
+}
+
+function addValueInImportedDict(
+  key: string,
+  val: string,
+  importedDict: Record<string, Set<string>>
+) {
+  if (!importedDict[key]) {
+    importedDict[key] = new Set<string>();
+  }
+  importedDict[key].add(val);
+}
+
+function buildMethodParamMap(model: RLCModel): Map<string, OperationParameter> {
+  const map = new Map<string, OperationParameter>();
+  (model.parameters ?? []).forEach((p) => {
+    const operatonConcante = getOperationConcate(
+      p.operationName,
+      p.operationGroup,
+      model.options?.sourceFrom
+    );
+    map.set(operatonConcante, p);
+  });
+  return map;
+}
+
+function getOperationConcate(
+  opName: string,
+  opGroup: string,
+  sourceFrom?: string
+) {
+  return sourceFrom === "Swagger"
+    ? opGroup === "" || opGroup === "Client"
+      ? opName
+      : `${opGroup}${opName}`
+    : `${opGroup}_${opName}`;
```

Comment:
not sure why we should have underscore if it's sourceFrom TypeSpec ?

Reply:
the method `getOperationConcate` is just an inner private method to build the operation group plus name info for linkage usage. I added the underscore here is just for consistency with autorest logic https://github.com/Azure/autorest.typescript/blob/main/packages/autorest.typescript/src/utils/nameUtils.ts#L241.

but pls notice the underscore here will be translated into sperator when normalization.